bindings/cs/rl.net/Native/SenderAdapter.cs

```diff
@@ -0,0 +1,145 @@
+using System;
+using System.Threading;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Diagnostics;
+
+namespace Rl.Net.Native {
+    internal delegate IntPtr sender_create_fn(IntPtr configuration, [MarshalAs(UnmanagedType.FunctionPtr)]error_fn error_callback, IntPtr error_ctx);
+
+    internal delegate int sender_init_fn(IntPtr managed_handle, IntPtr status);
+
+    internal delegate int sender_send_fn(IntPtr managed_handle, IntPtr buffer, IntPtr status);
+
+    internal delegate void sender_release_fn(IntPtr managed_handle);
+
+    [StructLayout(LayoutKind.Sequential)]
+    internal struct sender_vtable
+    {
+        [MarshalAs(UnmanagedType.FunctionPtr)] 
+        public sender_init_fn init;
+
+        [MarshalAs(UnmanagedType.FunctionPtr)] 
+        public sender_send_fn send;
+
+        [MarshalAs(UnmanagedType.FunctionPtr)] 
+        public sender_release_fn release;
+    }
+
+    internal class SenderAdapter
+    {
+        private ISender senderImplementation;
+
+        public SenderAdapter(ISender senderImplementation)
+        {
+            this.senderImplementation = senderImplementation;
+        }
+
+        public static sender_vtable VTable
+        {
+            get;
+        } = CreateAdapterVTable();
+
+        private static sender_vtable CreateAdapterVTable()
+        {
+            return new sender_vtable
+            {
+                init = (managed_handle, status) => InvokeInit(managed_handle, status),
```

Comment:
We should not depend on the C# compiler caching those delegates in static variables and do it ourselves explicitly.

Those delegates are not explicitly tracked by managed code after a call uses `SenderAdapter.Vtable`.

Reply:
Nice catch!